components/search/filter-sidebar.js

```diff
@@ -119,21 +120,40 @@ class FilterSidebar extends React.Component {
         if (asnValue && asnValue.match(asnRegEx) === null) {
           this.setState({
             asnError: 'Valid formats: AS1234, 1234',
-            [filterName]: e.target.value,
             isFilterDirty: false
           })
-          return
         } else {
           this.setState({
-            asnError: false
+            asnError: false,
+            isFilterDirty: true
           })
         }
         break
+      case 'domainFilter':
+        var domainValue = e.target.value
+        // eslint-disable-next-line no-useless-escape
+        var domainRegEx = /^(http:\/\/www\.|https:\/\/www\.|http:\/\/|https:\/\/)?[a-z0-9]+([\-\.]{1}[a-z0-9]+)*\.[a-z]{2,5}(:[0-9]{1,5})?(\/)?$/
```

Comment:
The `http://` or `https://` prefix must not be present for it to be a valid domain name.

Reply:
True. Can we not accept values with valid `http://` or `https://` and truncate them before passing on to the API? I am guessing people might copy paste such values often and end up thinking there is no data about them.

Reply:
As we are changing the behaviour to accepting domains, we should rather show a validation error if the user does not enter a valid domain and maybe in the validation error mention that it should not have the `http` prefix

Reply:
The validation error component doesn't work very well with longer messages. It already looks unpleasant this way.
![image](https://user-images.githubusercontent.com/700829/64191926-92a81680-ce47-11e9-99cc-99ae37a8b393.png)

We should be able to handle this better when we do https://github.com/ooni/design-system/issues/62

Reply:
Updated regex to validate for domain names without prefixes.